src/vm/builtins/builtin_runner/range_check.zig

```diff
@@ -216,45 +217,51 @@ pub const RangeCheckBuiltinRunner = struct {
         return pointer;
     }
 
-    /// Creates Validation Rules ArrayList
+    /// Creates Validation Rule in Memory
     ///
     /// # Parameters
     ///
     /// - `memory`: A `Memory` pointer of validation rules segment index.
-    /// - `address`: A `Relocatable` pointer to the validation rule.
     ///
-    /// # Returns
+    /// # Modifies
     ///
-    /// An `ArrayList(Relocatable)` containing the rules address
-    /// verification fails.
-    pub fn rangeCheckValidationRule(memory: *Memory, address: Relocatable) ?[]const Relocatable {
-        const num = ((memory.get(address) catch {
-            return null;
-        }) orelse {
-            return null;
-        }).tryIntoFelt() catch {
-            return null;
-        };
-
-        if (@bitSizeOf(u256) - @clz(num.toInteger()) <= N_PARTS * INNER_RC_BOUND_SHIFT) {
-            // TODO: unit tests
-            return &[_]Relocatable{address};
-        } else {
-            return null;
-        }
+    /// - `memory`: Adds validation rule to `memory`.
+    pub fn addValidationRule(self: *const Self, memory: *Memory) !void {
+        try memory.addValidationRule(@intCast(self.base), rangeCheckValidationRule);
     }
 
-    /// Creates Validation Rule in Memory
+    /// Returns the min and max values of range check
     ///
     /// # Parameters
     ///
-    /// - `memory`: A `Memory` pointer of validation rules segment index.
+    /// - `memory`: A `Memory` pointer.
     ///
-    /// # Modifies
+    /// # Returns
     ///
-    /// - `memory`: Adds validation rule to `memory`.
-    pub fn addValidationRule(self: *const Self, memory: *Memory) !void {
-        try memory.addValidationRule(@intCast(self.base), rangeCheckValidationRule);
+    /// - An `Array`containing the min and max of range check.
+    pub fn getRangeCheckUsage(self: *Self, memory: *Memory) ?[2]usize {
```

Comment:
I think that it is better to use a tuple here for the result, what do you think?

```zig
std.meta.Tuple(&.{ usize, usize })
```

You can find examples at other places of the code or here if needed: https://ziglang.org/documentation/master/#Tuples